projects/observability/src/shared/components/bar-gauge/bar-gauge.component.ts

```diff
@@ -0,0 +1,51 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { getPercentage, TypedSimpleChanges } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-bar-gauge',
+  styleUrls: ['./bar-gauge.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="bar-gauge">
+      <div *ngIf="this.title" class="title">{{ this.title | htDisplayTitle }}</div>
+      <div class="count">
+        {{ this.value?.toLocaleString() }} / {{ this.maxValue?.toLocaleString() }}
+        <span class="units" *ngIf="this.units">{{ this.units }}</span>
+      </div>
+      <div class="bar">
+        <div class="max-value-bar"></div>
+        <div
+          class="current-value-bar"
+          [ngClass]="{ 'almost-max-value': this.almostMaxValue, 'over-max-value': this.overMaxValue }"
+          [style.width.%]="this.valuePercentage"
+        ></div>
+      </div>
+    </div>
+  `
+})
+export class BarGaugeComponent implements OnChanges {
+  @Input()
+  public title?: string;
+
+  @Input()
+  public value?: number;
```

Comment:
Your use case will need multiple segments, right? Fine if you want to do it iteratively, but will break the current API.

Reply:
Yeah, true. I'll need to add those as well as a legend.

Actually, let me fix the multiple values now. I'll do the legend in a follow up.

Reply:
For multiple values, can the api be similar to the ht-gauge(For consistency)?

Reply:
I made it pretty close to ht-guage. I didn't wan the user to have to calculate min and max of each segment though, so that's a little different here.